### PR TITLE
Packed instruction/circuit flags

### PR DESCRIPTION
**Pull request by moodlezoup**
Replaces the `opcode` bytecode polynomial with `bitflags`, which contains circuit and instruction flags packed into a single field element per CPU step
TODO: enforce consistency between bitflags <> instruction_flags, circuit_flags

### CODE REVIEW

File: jolt-core/src/r1cs/constraints.rs [hunk 1]
```diff
@@ -336,9 +336,9 @@ impl<F: PrimeField> R1CSBuilder<F> {
 
         // Parse the input indices 
         let opcode = GET_INDEX(InputType::ProgVRW, 0);
-        let rs1 = GET_INDEX(InputType::ProgVRW, 1);
-        let rs2 = GET_INDEX(InputType::ProgVRW, 2);
-        let rd = GET_INDEX(InputType::ProgVRW, 3);
+        let rd = GET_INDEX(InputType::ProgVRW, 1);
+        let rs1 = GET_INDEX(InputType::ProgVRW, 2);
+        let rs2 = GET_INDEX(InputType::ProgVRW, 3);
```
Comment (moodlezoup) [inline jolt-core/src/r1cs/constraints.rs:341]: keeps the register ordering consistent with `bytecode.rs`

**Review by sragss — APPROVED** [review]
LGTM

File: jolt-core/src/jolt/vm/bytecode.rs [hunk 2]
```diff
@@ -81,185 +80,73 @@ impl ELFRow {
...
     /// MLE of read/write addresses. For offline memory checking, each read is paired with a "virtual" write,
     /// so the read addresses and write addresses are the same.
     a_read_write: DensePolynomial<F>,
     /// MLE of read/write values. For offline memory checking, each read is paired with a "virtual" write,
-    /// so the read values and write values are the same. There are multiple values (opcode, rd, rs1, etc.)
-    /// associated with each memory address, so `v_read_write` comprises multiple polynomials.
-    v_read_write: FiveTuplePoly<F>,
+    /// so the read values and write values are the same. There are five values (bitflags, rd, rs1, rs2, imm)
+    /// associated with each memory address, so `v_read_write` comprises five polynomials.
+    v_read_write: [DensePolynomial<F>; 5],
```
[the remark this answers is not in the document]
Comment (moodlezoup) [reply jolt-core/src/jolt/vm/bytecode.rs:145]: the reason I changed it is that I kept running into footguns where we had `rd, rs1, rs2` in one place and `rs1, rs2, rd` in another

File: jolt-core/src/jolt/vm/bytecode.rs [hunk 3]
```diff
@@ -81,185 +80,73 @@ impl ELFRow {
...
-            .map(|x| if x.is_zero() { false } else { true })
-            .collect();
-
-        println!("opcode as u8: {:?}", (self.opcode as u8));
-        println!(
-            "opcode: {:?}",
-            RV32IM::from_repr(self.opcode as u8).unwrap()
-        );
-        println!("bits: {:?}", circuit_flags_bits);
+    pub fn bitflags<InstructionSet>(instruction: &ELFInstruction) -> u64
```
[the remark this answers is not in the document]
Comment (moodlezoup) [reply jolt-core/src/jolt/vm/bytecode.rs:91]: Done in #201 because I was working on the wrong branch 🤡

**Pull request merged**